src/random-seed/random-seed.c

```diff
@@ -306,6 +309,71 @@ static int save_seed_file(
         return 0;
 }
 
+static int refresh_boot_seed(void) {
+        uint8_t buffer[RANDOM_EFI_SEED_SIZE];
+        struct sha256_ctx hash_state;
+        _cleanup_free_ void *seed_file_bytes = NULL;
+        _cleanup_free_ char *esp_path = NULL;
+        _cleanup_close_ int seed_fd = -1;
+        size_t len;
+        ssize_t r;
+
+        assert_cc(RANDOM_EFI_SEED_SIZE == SHA256_DIGEST_SIZE);
+
+        /* We're doing this opportunistically, so if the seeding dance before didn't manage to initialize the
+         * RNG, there's no point in doing it here. Secondly, getrandom(GRND_NONBLOCK) has been around longer
+         * than EFI seeding anyway, so there's no point in having non-getrandom() fallbacks here. So if this
+         * fails, just return early to cut our losses. */
+        r = getrandom(buffer, sizeof(buffer), GRND_NONBLOCK);
+        if (r < 0 && errno == EAGAIN)
+                return log_debug_errno(-errno, "Random pool not initialized yet, so skipping EFI seed update");
+        else if (r < 0)
+                return log_error_errno(-errno, "Failed to generate random bytes for EFI seed: %m");
+        assert(r == sizeof(buffer));
+        sha256_init_ctx(&hash_state);
+        sha256_process_bytes(&r, sizeof(r), &hash_state);
+        sha256_process_bytes(buffer, r, &hash_state);
+
+        r = find_esp_and_warn(NULL, NULL, /* unprivileged_mode= */ false, &esp_path,
+                              NULL, NULL, NULL, NULL, NULL);
+        if (r == -ENOKEY)
+                return log_debug_errno(r, "Couldn't find any ESP, so not updating ESP random seed.");
+        else if (r < 0)
+                return r; /* find_esp_and_warn() already logged */
+
+        /* Read the existing seed, but if it's not there, return early, since it means bootctl hasn't yet set
+         * up random seeds for the loader. */
+        seed_fd = chase_symlinks_and_open("/loader/random-seed", esp_path, CHASE_PREFIX_ROOT,
```

Comment:
(btw, once  #25355 has landed this should also get CHASE_PROHIBIT_SYMLINKS)

Reply:
Okay, will keep my eye on that and leave this "unresolved" until that's merged.

Reply:
Done.